libevmasm/Assembly.cpp

```diff
@@ -675,22 +874,39 @@ LinkerObject const& Assembly::assemble() const
 				"Some immutables were read from but never assigned, possibly because of optimization."
 			);
 
-	if (!m_subs.empty() || !m_data.empty() || !m_auxiliaryData.empty())
+	if (!eof && (!m_subs.empty() || !m_data.empty() || !m_auxiliaryData.empty()))
 		// Append an INVALID here to help tests find miscompilation.
 		ret.bytecode.push_back(static_cast<uint8_t>(Instruction::INVALID));
 
-	for (auto const& [subIdPath, bytecodeOffset]: subRef)
+	auto const dataStart = ret.bytecode.size();
+
+	// TODO: this should better be done separately on develop.
```

Comment:
@nikola-matic This is actually already an improved version of https://github.com/ethereum/solidity/pull/13804 that does use bytecode hashes.